src/persist/src/indexed/mod.rs

```diff
@@ -182,28 +180,36 @@ impl Pending {
 ///   storage, and clears and responds to all pending responses.
 /// - Pending writes, seals, and allow_compactions are drained before processing
 ///   any other type of request.
+/// - When evaluating a request, the work of updating the state is given to
+///   AppliedState (which has no knowledge of storage, etc). Then, if this was
+///   successful, Indexed will serialize AppliedState and durably write it down.
 #[derive(Debug)]
 pub struct Indexed<L: Log, B: Blob> {
-    next_stream_id: Id,
-    saved_seqno: SeqNo,
-    highest_assigned_seqno: SeqNo,
-    // This is conceptually a map from `String` -> `Id`, but lookups are rare
-    // and this representation is optimized for the metadata serialization path,
-    // which is less rare.
-    id_mapping: Vec<StreamRegistration>,
-    graveyard: Vec<StreamRegistration>,
     // NB: we are not using Log for anything at the moment and instead have
     // all writes going directly to trace. At some point we'll need to revisit
     // what we want to do with Log, and whether we want it to live inside of
     // Indexed or somewhere else.
     log: L,
     blob: BlobCache<B>,
     maintainer: Maintainer<B>,
-    unsealeds: BTreeMap<Id, Unsealed>,
-    traces: BTreeMap<Id, Trace>,
     listeners: HashMap<Id, Vec<ListenFn<Vec<u8>, Vec<u8>>>>,
     metrics: Metrics,
+    state: AppliedState,
     pending: Pending,
+}
+
+#[derive(Debug)]
+struct AppliedState {
```

Comment:
perhaps I'm missing something silly but why is it called `AppliedState`? Is it supposed to mean like "this is state that various commands have been applied to, but it has not yet been persisted to storage?"

Reply:
ok found my answer

```
/// The cumulative state that results from applying some prefix of the persist
/// state change log.
```

Reply:
I don't want to do a name bikeshed but for me something like `Meta`, or `MemMeta` (to highlight that its the same data as `BlobMeta` but optimized for in-memory usage) is more evocative for me?

but really the name is nbd for me so feel free to take it or leave it.

Reply:
Hah, I was actually thinking of going the other way and renaming BlobMeta to match this. Let's hop on a zoom today and brainstorm on naming.

I put the rustdoc on the wrong block? :facepalm: